File: app/src/main/java/com/example/waveplayer/Song.java
package com.example.waveplayer;

import androidx.annotation.Nullable;

import com.example.waveplayer.random_playlist.AudioUri;

import java.io.Serializable;

public class Song implements Comparable<Song>, Serializable {

    public final Long id;

    public final String title;

    private boolean selected;

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public Song(Long id, String title){
        this.id = id;
        this.title = title;
    }

    @Override
    public int compareTo(Song o) {
        return title.compareTo(o.title);
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return obj instanceof Song && id.equals(((Song) obj).id);
    }

    @Override
    public int hashCode() {
        return System.identityHashCode(this);
    }

    public static Song getSong(AudioUri audioUri){
        return new Song(audioUri.id, audioUri.title);
    }

}

File: app/src/main/java/io/fourthFinger/pinkyPlayer/random_playlist/Song.kt
package io.fourthFinger.pinkyPlayer.random_playlist

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable

/**
 * A selectable song.
 *
 * @param id The unique id of the song.
 * @param title The title of the song.
 */
@Entity(tableName = "songs")
class Song(
    @field:PrimaryKey val id: Long,
    @field:ColumnInfo(name = "title") val title: String
) : Comparable<Song>, Serializable {

    @Ignore @Volatile
    private var selected = false

    /**
     * Gets the selection status of this song.
     *
     * @return The selection status of this song.
     */
    @Synchronized
    fun isSelected(): Boolean {
        return selected
    }

    /**
     * Sets this song's selection status.
     *
     * @param selected Whether to set this songs to selected.
     */
    @Synchronized
    fun setSelected(selected: Boolean) {
        this.selected = selected
    }

    override fun compareTo(other: Song): Int {
        return title.compareTo(other.title)
    }

    override fun equals(other: Any?): Boolean {
        return other is Song && title == (other).title
    }

    override fun hashCode(): Int {
        return title.hashCode()
    }

}